### PR TITLE
Update pyproject.toml - replace `output_formatting` with `output-formatting`

### LINKED ISSUE #55290: BUILD: pandas[output_formatting] doesn't install optional dependencies
### Installation check

- [X] I have read the [installation guide](https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html#installing-pandas).

### Platform

win-64

### Installation Method

pip install

### pandas Version

2.1.1

### Python Version

3.10.11

### Installation Logs

`pip install pandas[output_formatting]` doesn't install jinja2 and tabulate, as shown in the installation guide's [visualization table](https://pandas.pydata.org/docs/getting_started/install.html#visualization).  That table has only matplotlib for optional dependencies to plot, but `pip install pandas[plot]` actually includes more packages.

### PR DESCRIPTION
**Pull request by barseghyanartur**
closes #55290

Simply said, none of this would work (dependencies from `output-formatting` would not be installed):

```
pip install pandas[output-formatting]
pip install pandas[output_formatting]
pip install pandas[output-formatting,xml]
```

When being distributed, underscores are replaces with dashes. So, in the packaged PKG-INFO file you won't find spots of `output_formatting`, but only `output-formatting`:

```
Requires-Dist: lxml>=4.8.0; extra == "xml"
Requires-Dist: jinja2>=3.1.2; extra == "output-formatting"
Requires-Dist: tabulate>=0.8.10; extra == "output-formatting"
```

But on the same time, this:

```
Provides-Extra: output_formatting
```

This fix addresses that.

### CODE REVIEW

**Comment by mroeschke** [comment]
Thanks could you add a whatsnew note in `2.1.2.rst`?

**Comment by barseghyanartur** [comment]
@mroeschke:

Done.

File: doc/source/whatsnew/v2.1.2.rst [hunk 1]
```diff
@@ -29,11 +29,12 @@ Bug fixes
...
--
+- Fixed non-working installation of optional dependency group ``output_formatting``. Replacing underscore ``_`` with a dash ``-`` fixes broken dependency resolution. A correct way to use now is ``pip install pandas[output-formatting]``.
 -
 
 .. ---------------------------------------------------------------------------
 .. _whatsnew_212.contributors:
 
 Contributors
 ~~~~~~~~~~~~
+.. contributors:: v2.1.1..v2.1.2|HEAD
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.2.rst:40]:
```suggestion
```

**Comment by mroeschke** [comment]
Could you also change the references in `doc/source/getting_started/install.rst` and `.github/workflows/package-checks.yml`?

**Comment by barseghyanartur** [comment]
@mroeschke:

Done.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @barseghyanartur

**Pull request merged**